### PR TITLE
Release ESMFold IPA only models

### PR DESCRIPTION
**Pull request by ebetica**
See table S1, these are a set of models that were used to benchmark the capabilities of the language model.

### CODE REVIEW

**Review by tomsercu — APPROVED** [review]
LGTM!
Two things (a) add new models to hubconf.py
(b) add summary of new models to README table

File: hubconf.py [hunk 1]
```diff
@@ -6,27 +6,38 @@
...
     esm2_t6_8M_UR50D,
     esm2_t12_35M_UR50D,
     esm2_t30_150M_UR50D,
     esm2_t33_650M_UR50D,
     esm2_t36_3B_UR50D,
     esm2_t48_15B_UR50D,
+    esm_if1_gvp4_t16_142M_UR50,
+    esm_msa1_t12_100M_UR50S,
+    esm_msa1b_t12_100M_UR50S,
+    esmfold_ipa_8M,
```
Comment (tomsercu) [inline hubconf.py:30]: As discussed async - let's rename to `esmfold_structure_module_only_*`, for clarity

**Review by tomsercu — CHANGES_REQUESTED** [review]
Blocked on renaming and errant print statements

File: esm/esmfold/v1/esmfold.py [hunk 2]
```diff
@@ -153,24 +188,35 @@ def forward(
...
         esm_s = esm_s.detach()
 
         # === preprocessing ===
         esm_s = (self.esm_s_combine.softmax(0).unsqueeze(0) @ esm_s).squeeze(2)
 
         s_s_0 = self.esm_s_mlp(esm_s)
-        s_z_0 = s_s_0.new_zeros(B, L, L, self.cfg.trunk.pairwise_state_dim)
+        if self.cfg.use_esm_attn_map:
+            esm_z = esm_z.to(self.esm_s_combine.dtype)
+            esm_z = esm_z.detach()
```
[the remark this answers is not in the document]
Comment (ebetica) [reply esm/esmfold/v1/esmfold.py:205]: I'm just following the above when I do this :) I think we should keep it consistent.

File: esm/esmfold/v1/pretrained.py [hunk 3]
```diff
@@ -52,3 +54,125 @@ def esmfold_v1():
...
+
+def esmfold_ipa_8M():
+    """
+    ESMFold baseline model using 8M ESM-2, 0 folding blocks.
+    ESM-2 here is trained out to 500K updates.
+    This is a model designed to test the capabilities of the language model
+    when ablated for number of parameters in the language model.
+    See table S1 in (Lin et al, 2022).
+    """
+    return _load_model("esmfold_ipa_8M")
```
[the remark this answers is not in the document]
Comment (ebetica) [reply esm/esmfold/v1/pretrained.py:67]: Again, just mirroring what was done before. @rmrao was there any good reason?

File: hubconf.py (hunk 1 above)
Comment (ebetica) [reply hubconf.py]: Done

File: esm/esmfold/v1/esmfold.py [hunk 4]
```diff
@@ -58,6 +75,13 @@ def __init__(self, esmfold_config=None, **kwargs):
             nn.ReLU(),
             nn.Linear(c_s, c_s),
         )
+        if cfg.use_esm_attn_map:
```
Comment (nikitos9000) [inline esm/esmfold/v1/esmfold.py:78]: Will it work without adding the respective field to ESMFoldConfig?

Comment (ebetica) [reply esm/esmfold/v1/esmfold.py:78]: They should be properly added.

**Review by tomsercu — APPROVED** [review]
Great confirmed all works - let's merge! 
Thanks Zeming!!

**Review by nikitos9000 — APPROVED** [review]
Looks good!

**Pull request merged**